pkg/pool-weighted/test/managed/BasePool.test.ts

```diff
@@ -0,0 +1,1000 @@
+        it('do not revert', async () => {
+          await expect(normalExit()).to.not.be.reverted;
+        });
+
+        it('calls inner onExit hook with exit parameters', async () => {
+          const receipt = await normalExit();
+          expectEvent.inIndirectReceipt(receipt, pool.interface, 'InnerOnExitPoolCalled', {
+            sender: sender.address,
+            balances: initialBalances,
+            userData: defaultAbiCoder.encode(['uint256'], [OTHER_EXIT_KIND]),
+          });
+        });
+
+        it('returns the output of the inner onExit hook and 0 due protocol fees', async () => {
+          const onExitReturn = await pool
+            .connect(vaultSigner)
+            .callStatic.onExitPool(poolId, sender.address, recipient.address, initialBalances, 0, 0, '0x');
+          expect(onExitReturn).to.be.deep.eq([
+            Array(tokens.length).fill(await pool.ON_EXIT_RETURN()),
+            Array(tokens.length).fill(bn(0)),
+          ]);
+        });
+      });
+    }
+  });
+
+  describe('pool initialization', () => {
+    let pool: Contract;
+    let sender: SignerWithAddress, recipient: SignerWithAddress;
+    let poolId: string, userData: string;
+    let request: JoinPoolRequest;
+    let initialBalances: Array<BigNumber>;
+
+    sharedBeforeEach('set up pool and initial join request', async () => {
+      sender = poolOwner;
+      recipient = other;
+      pool = await deployBasePool({
+        pauseWindowDuration: PAUSE_WINDOW_DURATION,
+      });
+      poolId = await pool.getPoolId();
+
+      const initialBalancePerToken = 1000;
+
+      await tokens.mint({ to: sender, amount: fp(initialBalancePerToken) });
+      await tokens.approve({ from: sender, to: vault });
+
+      initialBalances = Array(tokens.length).fill(fp(initialBalancePerToken));
+      userData = WeightedPoolEncoder.joinInit(initialBalances);
+      request = {
+        assets: tokens.addresses,
+        maxAmountsIn: initialBalances,
+        userData,
+        fromInternalBalance: false,
+      };
+    });
+
+    context('when paused', () => {
+      sharedBeforeEach(async () => {
+        await authorizer
+          .connect(admin)
+          .grantPermissions([await actionId(pool, 'pause')], sender.address, [ANY_ADDRESS]);
+        await pool.connect(sender).pause();
+      });
+
+      it('reverts', async () => {
+        await expect(
+          vault.connect(sender).joinPool(poolId, sender.address, recipient.address, request)
+        ).to.be.revertedWith('PAUSED');
+      });
+    });
+
+    context('when not paused', () => {
+      it('calls inner initialization hook', async () => {
+        const receipt = await (
+          await vault.connect(sender).joinPool(poolId, sender.address, recipient.address, request)
+        ).wait();
+
+        expectEvent.inIndirectReceipt(receipt, pool.interface, 'InnerOnInitializePoolCalled', {
+          userData,
+        });
+      });
+
+      it('burns minimum bpt', async () => {
```

Comment:
```suggestion
      it('locks the minimum bpt in the zero address', async () => {
```

This BPT is not burnt (i.e. destroyed), it is sent to the zero address so that it can never be claimed for tokens (and burnt), meaning the supply can never drop to zero again.